### PR TITLE
MODINREACH-155 Fix Create INN-Reach Item Hold Local Endpoint

### PR DESCRIPTION
**Pull request by maryna-steshenko**
## Purpose

Fix endpoint path, request DTO, request creation errors and make request creation asynchronous.
## Approach

- External services endpoints paths modified.
- Create item hold local endpoint modified.
- Request creation method made asynchronous.
- Field centralItemType deleted from request DTO.
- InnReachTransaction controller tests modified.

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionServiceImpl.java [hunk 1]
```diff
@@ -39,12 +47,20 @@ public InnReachResponseDTO createInnReachTransactionItemHold(String trackingId,
     var response = new InnReachResponseDTO();
     response.setStatus("ok");
     try {
-      centralServerService.getCentralServerByCentralCode(centralCode);
+      var centralServer = centralServerService.getCentralServerByCentralCode(centralCode);
       var transaction = createTransactionWithItemHold(trackingId, centralCode);
       var itemHold = mapper.toItemHold(dto);
+      var item = inventoryClient.getItemByHrId(itemHold.getItemId());
+      var materialType = materialRepository.findOneByCentralServerIdAndMaterialTypeId(
+        centralServer.getId(), item.getMaterialType().getId()).orElseThrow(
```
Comment (oliinyko) [inline src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionServiceImpl.java]: add variables for centralServer.getId(), item.getMaterialType().getId()

File: src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionServiceImpl.java [hunk 2]
```diff
@@ -39,12 +47,20 @@ public InnReachResponseDTO createInnReachTransactionItemHold(String trackingId,
...
+        () -> new EntityNotFoundException("Material type mapping for central server id = " + centralServer.getId()
+          + " and material type id = " + item.getMaterialType().getId() + " not found")
+      );
+      itemHold.setCentralItemType(materialType.getCentralItemType());
       transaction.setHold(itemHold);
       repository.save(transaction);
     } catch (Exception e) {
+      log.warn(e.getMessage());
```
Comment (oliinyko) [inline src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionServiceImpl.java]: add a message with some context and pass the whole exception as an argument

File: pom.xml [hunk 3]
```diff
@@ -197,6 +198,12 @@
       <artifactId>marc4j</artifactId>
       <version>${marc4j.version}</version>
     </dependency>
+    <dependency>
+      <groupId>org.awaitility</groupId>
```
Comment (dmtkachenko) [inline pom.xml:209]: I don't see any tests that use Awaitility. Are they missing in PR?

Comment (maryna-steshenko) [reply pom.xml:209]: Awaitility is used in `InnReachTransactionControllerTest` in such tests as `return200HttpCode_and_sendRequest_whenItemHoldTransactionCreated`,  `issueOwningSideCancelsRequest_when_createInnReachTransaction_and_creatingRequestFails` and `issueOwningSideCancelsRequest_when_createInnReachTransaction_and_itemIsNotRequestable`. You can find its usages by looking for method `await()` in these tests.

File: src/main/resources/swagger.api/schemas/transactionItemHoldDTO.json [hunk 4]
```diff
@@ -41,12 +41,6 @@
       "description": "Need before time",
       "type": "integer"
     },
-    "centralItemType": {
-      "description": "Central item type",
-      "type": "integer",
-      "minimum" : 0,
-      "maximum" : 255
-    },
```
Comment (oliinyko) [inline src/main/resources/swagger.api/schemas/transactionItemHoldDTO.json:49]: why is it removed?

File: src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionServiceImpl.java [hunk 5]
```diff
@@ -39,12 +47,22 @@ public InnReachResponseDTO createInnReachTransactionItemHold(String trackingId,
...
       var transaction = createTransactionWithItemHold(trackingId, centralCode);
       var itemHold = mapper.toItemHold(dto);
+      var item = inventoryClient.getItemByHrId(itemHold.getItemId());
+      var materialTypeId = item.getMaterialType().getId();
+      var materialType = materialRepository.findOneByCentralServerIdAndMaterialTypeId(
+        centralServerId, materialTypeId).orElseThrow(
+        () -> new EntityNotFoundException("Material type mapping for central server id = " + centralServerId
+          + " and material type id = " + materialTypeId + " not found")
+      );
+      itemHold.setCentralItemType(materialType.getCentralItemType());
```
Comment (oliinyko) [inline src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionServiceImpl.java]: let's move this logic to MaterialTypeMappingService

File: src/main/resources/swagger.api/schemas/transactionItemHoldDTO.json (hunk 4 above)
Comment (maryna-steshenko) [reply src/main/resources/swagger.api/schemas/transactionItemHoldDTO.json:49]: This parameter is not provided in a request, it was added to the dto schema by a mistake.

File: pom.xml (hunk 3 above)
Comment (dmtkachenko) [reply pom.xml:209]: yes, I see, thanks

**Review by dmtkachenko — APPROVED** [review]

**Review by oliinyko — APPROVED** [review]

**Pull request merged**